config-model/src/test/java/com/yahoo/schema/expressiontransforms/BooleanExpressionTransformerTestCase.java

```diff
@@ -33,25 +37,48 @@ public void testIt() throws Exception {
         assertTransformed("if(1 - 1, true, 1 - 1)", "1 - 1 || 1 - 1");
     }
 
+    @Test
+    public void testNotSkewingNonBoolean() throws Exception {
+        assertTransformed("a + b + c * d + e + f", "a + b + c * d + e + f");
+        var expr = new BooleanExpressionTransformer()
+                .transform(new RankingExpression("a + b + c * d + e + f"),
+                        new TransformContext(Map.of(), new MapTypeContext()));
+        assertTrue(expr.getRoot() instanceof ArithmeticNode);
+        ArithmeticNode root = (ArithmeticNode) expr.getRoot();
+        assertEquals(5, root.operators().size());
+        assertEquals(6, root.children().size());
+    }
+
+    @Test
+    public void testTransformPreservesPrecedence() throws Exception {
+        assertUnTransformed("a");
+        assertUnTransformed("a + b");
+        assertUnTransformed("a + b + c");
+        assertUnTransformed("a * b");
+        assertUnTransformed("a + b * c + d");
+        assertUnTransformed("a + b + c * d + e + f");
+        assertUnTransformed("a * b + c + d + e * f");
+        assertUnTransformed("(a * b) + c + d + e * f");
+        assertUnTransformed("(a * b + c) + d + e * f");
+        assertUnTransformed("a * (b + c) + d + e * f");
+        assertUnTransformed("(a * b) + (c + (d + e)) * f");
+    }
+
+    private void assertUnTransformed(String input) throws Exception {
+        assertTransformed(input, input);
+    }
+
     private void assertTransformed(String expected, String input) throws Exception {
         var transformedExpression = new BooleanExpressionTransformer()
                                             .transform(new RankingExpression(input),
                                                        new TransformContext(Map.of(), new MapTypeContext()));
 
         assertEquals(new RankingExpression(expected), transformedExpression, "Transformed as expected");
 
-        MapContext context = contextWithSingleLetterVariables();
```

Comment:
Not really any point of evaluating at all without this since all expressions will just be a bunch of operations on zeros, which will all come out as zero.

Reply:
I suspected so :) I removed since it was not used.
Now it is actually used too.